### ISSUE TITLE
operational?

### ISSUE THREAD

**Issue by Rik44444** [CableIM/Cable-Android#7]
Hi, is this project operational? If so i would be a big fan! Lack of federation is Signal's main problem. 
Issues if operational:
- your website is not working
question if operational:
- will Cable federate with the Signal app?
Rik

**Comment by paride** [comment]
No, unfortunately this is dead since Signal removed all the federation related code from the app and server.

**Comment by kravietz** [comment]
@paride Does it mean they removed it from new (pre-fork) releases or used some other way to prevent fork development?

**Comment by paride** [comment]
@kravietz Yes the federation code has been removed from the pre-fork releases (client and server) a few years ago, and the codebases diverged very rapidly, so it was impossible for what basically was a 1-person side project to maintain a fork with a significant diff from upstream.

They didn't really care about forks **but** they didn't want the Signal name or icon to be used, and they didn't offer support or documentation on how to set up the server side infrastructure. What they fought was the redistribution of the binary Signal apk, and they didn't want people to recompile the app and use the recompiled ("unofficial") binary to connect to the official servers, claiming this broke the server ToS. Very annoying.

Anyway this project *did* work until it was abandoned. We all used it at the local hackerspace, but it also started spreading out to other people outside of my circles even before it got really advertised or published anywhere else than a private f-droid repository. I was really sorry to let it die.

**Comment by Rik44444** [comment]
what are your thoughts on [RiotX](https://play.google.com/store/apps/details?id=im.vector.riotx&hl=en_US) @paride ? Is it a good substitute for federated messaging?
